packages/certs-v5/lib/domains.js

```diff
@@ -66,4 +66,37 @@ function printDomains (domainsTable, msg) {
   }
 }
 
-module.exports = {waitForDomains, printDomains}
+async function waitForCertIssuedOnDomains(context, heroku) {
+  function certIssuedOrFailedForAllCustomDomains (domains) {
+    domains = domains.filter((domain) => domain.kind === 'custom')
+    return _.every(domains, (domain) => domain.acm_status === 'cert issued' || domain.acm_status === 'failed')
+  }
+
+  function someFailed (domains) {
+    domains = domains.filter((domain) => domain.kind === 'custom')
+    return _.some(domains, (domain) => domain.acm_status === 'failed')
+  }
+
+  function apiRequest (context, heroku) {
+    return heroku.get(`/apps/${context.app}/domains`)
+  }
+
+  let domains = await apiRequest(context, heroku)
+
+  if (!certIssuedOrFailedForAllCustomDomains(domains)) {
+    await cli.action('Waiting until the certificate is issued to all domains', (async function () {
+      let i = 0;
+      do {
+        await wait(60*1000)
```

Comment:
Waiting an entire minute between poll ticks seems like quite a long time unless we actually expect this operation to always take several minutes. Otherwise we could end up making users wait an unnecessarily long time.

Reply:
It states on the [docs](https://devcenter.heroku.com/articles/automated-certificate-management#view-your-certificate-status) that issuing a certificate could take between 45 and 60 minutes, that's why I've chosen to poll every minute.

Maybe we can reduce to half of that time

Reply:
Oh wow, nevermind then. One minute definitely seems appropriate in that case 😂

Reply:
The reality for LetsEncrypt + Heroku ACM in my experience working on MSNI, is that it is much faster. Once a minute probably is fine, though.

Reply:
It'd be trickier but we could also do a sort of exponential backoff here where we start by checking every 15 seconds but later check every minute max.

Reply:
Hmm that would be nice, I'll give it a shot on Monday morning

Reply:
I've added a small backoff function. It starts waiting 15 seconds and then goes up to 60